addons/sale_timesheet/models/product_template.py

```diff
@@ -45,32 +45,35 @@ def _compute_visible_expense_policy(self):
                 product_template.visible_expense_policy = visibility
         return super()._compute_visible_expense_policy()
 
-    @api.depends('service_tracking', 'service_policy', 'type', 'sale_ok')
     def _compute_product_tooltip(self):
-        super()._compute_product_tooltip()
-        for record in self.filtered(lambda record: record.type == 'service' and record.sale_ok):
-            if record.service_policy == 'delivered_timesheet':
-                if record.service_tracking == 'no':
-                    record.product_tooltip = _(
-                        "Invoice based on timesheets (delivered quantity) on projects or tasks "
-                        "you'll create later on."
-                    )
-                elif record.service_tracking == 'task_global_project':
-                    record.product_tooltip = _(
-                        "Invoice based on timesheets (delivered quantity), and create a task in "
-                        "an existing project to track the time spent."
-                    )
-                elif record.service_tracking == 'task_in_project':
-                    record.product_tooltip = _(
-                        "Invoice based on timesheets (delivered quantity), and create a project "
-                        "for the order with a task for each sales order line to track the time "
-                        "spent."
-                    )
-                elif record.service_tracking == 'project_only':
-                    record.product_tooltip = _(
-                        "Invoice based on timesheets (delivered quantity), and create an empty "
-                        "project for the order to track the time spent."
-                    )
+        service_products = self.filtered(lambda product: product.type == 'service' and product.sale_ok and self.service_policy == 'delivered_timesheet')
+        super(ProductTemplate, (self - service_products))._compute_product_tooltip()
+        for record in service_products:
+            record.product_tooltip = _(
+                "Invoice based on timesheets (delivered quantity)"
+            )
```

Comment:
Honestly, this is a nightmare for people working on our translations.
Please keep using a full string in each `if` instead of trying to extract a shared part ;).

Reply:
I did this to make use this sentence for event and other types
Like this:
![image](https://github.com/odoo/odoo/assets/77771734/26053184-d61b-432e-92a4-147d738b510e)
Checking with DALA he want to update `Event` and other types tooltip for different invoicing policy or not :see_no_evil:

Okay! I'll revert that :100: